packages/block-editor/src/utils/block-bindings.js

```diff
@@ -44,8 +48,11 @@ export function useBlockBindingsUtils() {
 	 * ```
 	 */
 	const updateBlockBindings = ( bindings ) => {
-		const { metadata } = getBlockAttributes( clientId );
-		const newBindings = { ...metadata?.bindings };
+		const {
+			metadata: { bindings: currentBindings, ...metadata },
+		} = getBlockAttributes( clientId );
+		const newBindings = { ...currentBindings };
```

Comment:
If I am not mistaken, this will break when there is no `metadata`and it is undefined. For example, when adding a binding to a new paragraph block.

Reply:
We might need to do something like this maybe:

```js
const { metadata: { bindings: currentBindings, ...metadata } = {} } =
    getBlockAttributes( clientId );
```

Reply:
Good catch:

<img width="488" alt="Screenshot 2024-08-26 at 14 17 26" src="https://github.com/user-attachments/assets/7e20087a-acd8-4efb-8dbc-e9eaae63c50b">

<img width="550" alt="Screenshot 2024-08-26 at 14 19 26" src="https://github.com/user-attachments/assets/e86191cc-ceb5-41eb-8e0c-ca89170c3aec">

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/64740/commits/9e98a29fc6f9aae646d74361a8d224d7fac291af